src/psyclone/psyir/transformations/loop_trans.py

```diff
@@ -99,16 +115,19 @@ def validate(self, node, options=None):
                 f"{[type(child).__name__ for child in node.children]}.")
 
         if not options:
-            options = {}
-        if not isinstance(options, dict):
-            raise TransformationError(
-                f"Transformation validate method 'options' argument must be a "
-                f"dictionary but found '{type(options).__name__}'.")
-
-        verbose = options.get("verbose", False)
+            self.validate_options(**kwargs)
+            verbose = self.get_option("verbose", **kwargs)
```

Comment:
Could/should we merge together `options` with **kwargs and then we could do away with this `if not options....else...`?

Reply:
Since we're going to deprecate `options`, please add a TODO here. I think @sergisiso was in favour of printing a deprecation warning too but presumably we can do that in the base class?

Reply:
I think it'll have to be on a class-by-class basis, those that call super().apply() can use the inherited call to their parent for the deprecation print. I'll add this into some of these methods for now but spreading it around the code more will be later in this PR.